api/apiFunctions.ts

```diff
@@ -301,3 +297,100 @@ export async function createEntry({
     throw new Error('Error creating entry');
   }
 }
+
+/**
+ * Retrieves a list of all leagues.
+ * @returns {ILeague[]} A list of all available leagues.
+ */
+export async function getAllLeagues(): Promise<IAllLeagues[]> {
+  try {
+    const response = await databases.listDocuments(
+      appwriteConfig.databaseId,
+      Collection.LEAGUE,
+    );
+
+    const allLeagues = response.documents.map((league) => ({
+      leagueId: league.$id,
+      leagueName: league.leagueName,
+      logo: '',
+      participants: league.participants,
+      survivors: league.survivors,
+    }));
+
+    return allLeagues;
+  } catch (error) {
+    console.error(error);
+    throw new Error('Error getting all leagues');
+  }
+}
+/**
+ * Retrieves the user document ID dynamically from the database.
+ * @param userId - The user ID
+ * @returns {string} - The user document ID
+ */
+export const getUserDocumentId = async (userId: string): Promise<string> => {
+  try {
+    const response = await databases.listDocuments(
+      appwriteConfig.databaseId,
+      Collection.USERS,
+      [Query.equal('userId', userId)],
+    );
+
+    if (response.documents.length === 0) {
+      throw new Error('User document not found');
+    }
+
+    // Assuming only one document per userId
+    return response.documents[0].$id;
+  } catch (error) {
+    console.error('Error getting user document ID:', error);
+    throw new Error('Error getting user document ID');
```

Comment:
Add `error` prop to this error for the stack trace.

Reply:
Updated